src/Provider/Doctrine/Auditing/Event/DoctrineSubscriber.php

```diff
@@ -32,32 +31,44 @@ public function __construct(TransactionManager $transactionManager)
      */
     public function onFlush(OnFlushEventArgs $args): void
     {
-        $entityManager = $args->getEntityManager();
+        $entityManager = method_exists($args, 'getObjectManager')
+            ? $args->getObjectManager()
+            : $args->getEntityManager();
+
         $transaction = new Transaction($entityManager);
+        // Populate transaction
+        $this->transactionManager->populate($transaction);
+
+        $driver = $entityManager->getConnection()->getDriver();
+        if ($driver instanceof DHDriver) {
+            $driver->addDHFlusher(function () use ($transaction): void {
+                $this->transactionManager->process($transaction);
+                $transaction->reset();
+            });
 
+            return;
+        }
+        trigger_deprecation('damienharper/auditor', '2.2', 'SQLLogger is deprecated. Use DHMiddleware instead');
```

Comment:
I don't really sure if we need this message

Reply:
Well, you can let it here. We'll remove it once auditor 3.0 is released.